app/controllers/application_controller.rb

```diff
@@ -130,7 +130,10 @@ def skip_session_expiration
   end
 
   def set_locale
-    I18n.locale = params[:locale] || I18n.default_locale
+    I18n.locale =
+      http_accept_language.compatible_language_from(I18n.available_locales) ||
+      params[:locale] ||
+      I18n.default_locale
```

Comment:
Also, we need to add a method to sanitize the URL parameter. If the URL parameter does not match an available locale, it will throw an error.